lib/pages/base/base_view_model.dart

```diff
@@ -1,3 +1,55 @@
+import 'package:domain/interactor/input/base_input.dart';
+import 'package:domain/interactor/output/output_observer.dart';
+import 'package:domain/usecase/base/base_use_case.dart';
 import 'package:flutter/material.dart';
 
-abstract class BaseViewModel extends ChangeNotifier {}
+abstract class BaseViewModel extends ChangeNotifier {
+  bool get isLoading => _isLoading;
+  bool _isLoading = false;
+  int _loadingCount = 0;
+
+  Future launchUseCase<I extends BaseInput, O>({
+    required BaseUseCase<I, O> useCase,
+    required I input,
```

Comment:
I tried but unfortunately ```AsyncValue``` doesn't have any apis to notify value itself and we can't reassign value of ```AsyncValue``` param because function param is final. So widget does not rebuild